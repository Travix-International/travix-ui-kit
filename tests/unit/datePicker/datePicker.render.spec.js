import { shallow } from 'enzyme';
import React from 'react';
import DatePicker from '../../../components/datePicker/datePicker';

describe('DatePicker', () => {
  describe('#render()', () => {
    const onChange = () => {};
    const valueFormatterFn = jest.fn();

    it('should render disabled DatePicker', () => {
      const wrapper = shallow(
        <DatePicker
          disabled
          name="date-picker"
          onChange={onChange}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render DatePicker with label', () => {
      const wrapper = shallow(
        <DatePicker
          label="label"
          name="date-picker"
          onChange={onChange}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render correct DatePicker with value', () => {
      const wrapper = shallow(
        <DatePicker
          name="date-picker"
          onChange={onChange}
          value="2017-03-20"
          valueFormatterFn={valueFormatterFn}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render open DatePicker', () => {
      const wrapper = shallow(
        <DatePicker
          name="date-picker"
          open
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
