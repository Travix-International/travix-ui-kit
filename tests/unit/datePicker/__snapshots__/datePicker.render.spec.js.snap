// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker #render() should render DatePicker with label 1`] = `
<div
  className="ui-datepicker"
>
  <label
    htmlFor="ui-datepicker-input-date-picker"
    id="ui-datepicker-label-date-picker"
  >
    label
  </label>
  <Input
    disabled={false}
    id="ui-datepicker-input-date-picker"
    mods={
      Array [
        "datepicker",
      ]
    }
    multiline={false}
    name="date-picker"
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
    value=""
  />
  <div
    aria-hidden={false}
    className="ui-datepicker__calendar-wrapper"
  >
    <Calendar
      multiplemode={true}
      onMouseDown={[Function]}
      onSelectDay={[Function]}
      selectionType="normal"
    />
  </div>
</div>
`;

exports[`DatePicker #render() should render correct DatePicker with value 1`] = `
<div
  className="ui-datepicker"
>
  <Input
    disabled={false}
    id="ui-datepicker-input-date-picker"
    mods={
      Array [
        "datepicker",
      ]
    }
    multiline={false}
    name="date-picker"
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
  />
  <div
    aria-hidden={false}
    className="ui-datepicker__calendar-wrapper"
  >
    <Calendar
      multiplemode={true}
      onMouseDown={[Function]}
      onSelectDay={[Function]}
      selectionType="normal"
    />
  </div>
</div>
`;

exports[`DatePicker #render() should render disabled DatePicker 1`] = `
<div
  className="ui-datepicker"
>
  <Input
    disabled={true}
    id="ui-datepicker-input-date-picker"
    mods={
      Array [
        "datepicker",
      ]
    }
    multiline={false}
    name="date-picker"
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
    value=""
  />
  <div
    aria-hidden={false}
    className="ui-datepicker__calendar-wrapper"
  >
    <Calendar
      multiplemode={true}
      onMouseDown={[Function]}
      onSelectDay={[Function]}
      selectionType="normal"
    />
  </div>
</div>
`;

exports[`DatePicker #render() should render open DatePicker 1`] = `
<div
  className="ui-datepicker ui-datepicker_active"
>
  <Input
    disabled={false}
    id="ui-datepicker-input-date-picker"
    mods={
      Array [
        "datepicker",
      ]
    }
    multiline={false}
    name="date-picker"
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
    value=""
  />
  <div
    aria-hidden={true}
    className="ui-datepicker__calendar-wrapper"
  >
    <Calendar
      multiplemode={true}
      onMouseDown={[Function]}
      onSelectDay={[Function]}
      selectionType="normal"
    />
  </div>
</div>
`;
